src/Silicon.cpp

```diff
@@ -0,0 +1,525 @@
+#ifdef DEBUGLOGGING
+        dbg<<"In Silicon::accumulate\n";
+        dbg<<"bounds = "<<b<<std::endl;
+        dbg<<"total nphotons = "<<photons.size()<<std::endl;
+        dbg<<"hasAllocatedWavelengths = "<<photons.hasAllocatedWavelengths()<<std::endl;
+        dbg<<"hasAllocatedAngles = "<<photons.hasAllocatedAngles()<<std::endl;
+        double Irr = 0.;
+        double Irr0 = 0.;
+        int zerocount=0, neighborcount=0, misscount=0;
+#endif
+
+        const int nx = b.getXMax() - b.getXMin() + 1;
+        const int ny = b.getYMax() - b.getYMin() + 1;
+        const int nxny = nx * ny;
+        dbg<<"nx,ny = "<<nx<<','<<ny<<std::endl;
+        _imagepolys.resize(nxny);
+        for (int i=0; i<nxny; ++i)
+            _imagepolys[i] = _emptypoly;
+        dbg<<"Built poly list\n";
+
+        const double invPixelSize = 1./_pixelSize; // pixels/micron
+        const double diffStep_pixel_z = _diffStep / (_sensorThickness * _pixelSize);
+
+        const int nphotons = photons.size();
+        std::vector<double> depth(nphotons);
+        calculateConversionDepth(photons, depth, ud);
+
+        GaussianDeviate gd(ud,0,1); // Random variable from Standard Normal dist.
+
+        // Start with the correct distortions for the initial image as it is already
+        updatePixelDistortions(target);
+
+        // Keep track of the charge we are accumulating on a separate image for efficiency
+        // of the distortion updates.
+        ImageAlloc<T> delta(b, 0.);
+
+        double addedFlux = 0.;
+        double next_recalc = _nrecalc;
+        for (int i=0; i<nphotons; i++) {
+            // Update shapes every _nrecalc electrons
+            if (addedFlux > next_recalc) {
+                updatePixelDistortions(delta.view());
+                target += delta;
+                delta.setZero();
+                next_recalc = addedFlux + _nrecalc;
+            }
+
+            // Get the location where the photon strikes the silicon:
+            double x0 = photons.getX(i); // in pixels
+            double y0 = photons.getY(i); // in pixels
+
+            double dz = depth[i];  // microns
+            if (photons.hasAllocatedAngles()) {
+                double dxdz = photons.getDXDZ(i);
+                double dydz = photons.getDYDZ(i);
+                double dz_pixel = dz * invPixelSize;
+                x0 += dxdz * dz_pixel; // dx in pixels
+                y0 += dydz * dz_pixel; // dy in pixels
+            }
+            // This is the reverse of depth. zconv is how far above the substrate the e- converts.
+            double zconv = _sensorThickness - dz;
+            if (zconv < 0.0) continue; // Throw photon away if it hits the bottom
+            // TODO: Do something more realistic if it hits the bottom.
+
+            // Now we add in a displacement due to diffusion
+            if (_diffStep != 0.) {
+                double diffStep = std::max(0.0, diffStep_pixel_z * (zconv - _pixelSize));
```

Comment:
I'm not sure about this line.  This used to be `zconv - 10.0`, and you switched the 10.0 to `_pixelSize`.  But why would you be subtracting the pixel size here?  I thought the 10.0 was just some kind of buffer to keep it from diffusing too much if zconv was close to 0.  

Like maybe once it got 10 microns from the surface, the electric fields we strong enough that it doesn't really diffuse anymore.  Or something like that anyway.  But I don't see how the pixel size would be relevant.  

So could you double check the math on this line?  Maybe the 10.0 here should be a different input parameter?  (Perhaps one that we don't have named yet?)